src/App.tsx

```diff
@@ -1,9 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const preparedQuery = query.toLowerCase().trim();
+  const visibleMovies = moviesFromServer.filter(({ title, description }) => {
+    const preparedTitle = title.toLowerCase();
+    const preparedDescription = description.toLowerCase();
+
+    const isIncludeTitle = preparedTitle.includes(preparedQuery);
+    const isIncludeDesription = preparedDescription.includes(preparedQuery);
```

Comment:
It seems that you can create a helper function to do `text1.toLowerCase().includes(text2.toLowerCase().trim())` stuff.